spin/cmds/meson.py

```diff
@@ -226,6 +230,14 @@ def _check_coverage_tool_installation(coverage_type: GcovReportFormat):
         )
 
 
+build_dir_option = click.option(
+    "-C",
+    "--build-dir",
+    default="build",
+    help="Meson build directory; package is installed into './{build-dir}-install'",
```

Comment:
This help text implies that subdirectories are not supported. And indeed, running
```sh
spin build -C build/testspin
spin python -C build/testspin
```
fails with
```plain
FileNotFoundError: No `site-packages` or `dist-packages` found under `build/testspin-install`
```

I'd definitely appreciate support for using a full path (sub directories, absolute path) here. That would allow targeting a generic `build/...` which is way easier to cover with `.gitignore` than `{build-dir}` which can have any name. It would also allow such cool things as
```
spin build -c /tmp/checkout-pr-feature
```

As it's currently implemented, this will quickly clog up valuable space in the top-level dir.

---

Another thought is to always use `build-intall` regardless of the `-C´ option and make meson update that directory with the cached build in `${build-dir}` when needed.

Reply:
Hmm, with the most recent version
```
spin build -C build/testspin
spin python -C build/testspin
```
still fails with the same error:
```plain
Traceback (most recent call last):
  File "/home/lg/.local/lib/micromamba/envs/skidev312/lib/python3.12/site-packages/spin/cmds/meson.py", line 738, in python
    p = _set_pythonpath(build_dir)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lg/.local/lib/micromamba/envs/skidev312/lib/python3.12/site-packages/spin/cmds/meson.py", line 100, in _set_pythonpath
    site_packages = _get_site_packages(build_dir)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lg/.local/lib/micromamba/envs/skidev312/lib/python3.12/site-packages/spin/cmds/meson.py", line 160, in _get_site_packages
    raise FileNotFoundError(
FileNotFoundError: No `site-packages` or `dist-packages` found under `build/testspin-install`
```

It works for an absolute path such as `/tmp/testspin` though! Did you test the relative path successfully on your side? Then it might be something with my setup..

Reply:
Is the leading `..` in `--destdir ../build/testspin-install` the culprit perhaps? (I updated the above traceback to include the full output)

Reply:
Ah, you want nested relative dirs; I'll fix that use case too.